design-proposals/memory-hotplug.md

```diff
@@ -0,0 +1,209 @@
+  dynamically scale according to the current number of enabled memory
+* Set the memory increment (new_guest - old_guest) for online memory expansion An integer multiple of 128 MB (operating
+  system requirements)
+* Memory hotplug depends on numa, VMs will have to be `NUMAable` to enable Memory hotplug
+
+#### API Examples
+
+##### VM Spec API
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: vm-cirros
+  name: vm-cirros
+spec:
+  running: false
+  liveUpdateFeatures:
+  +   memory
++     maximumGuest: 4Gi
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-cirros
+    spec:
+      domain:
+        memory:
+          guest: 4Gi
+```
+
+##### VMI Spec API
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vm-cirros
+  name: vm-cirros
+spec:
+  domain:
+    memory:
++     maxGuest: 10Gi
+  guest: 8Gi
+
+```
+
+## Status API
+
+The VMI status must indicate whether Memory hotplug is taking place. As the hotplug process begins the current state
+of `spec.domain.memory` needs to be propagated back to the VM/VMI `status.currentMemory` to allow the user to have an
+indication of the hotplug processing.
+
+```
+	// Memory allow specified the detailed inside the vmi.
+		CurrentMemory *Memory `json:"currentMemory,omitempty"`
+```
+
+Conditions on the VM and VMI should indicate the status of the pending Memory hotplug actions. When errors occur, the
+conditions should propagate the errors to the VM.Status.Conditions field so users can receive feedback into the status
+of their declared state.
+
+```
+// Indicates that the VMI is in progress of Hot memory Plug/UnPlug
+VirtualMachineInstanceMemoryChange VirtualMachineInstanceConditionType = "HotMemoryChange"
+```
+
+Upon succesful hotplug action the condition will be removed. Otherwise, a human readable reason and message indicating
+whether the hotplug change is pending or encountering errors.
+
+## Extensibility
+
+We will be able to use the `VMLiveUpdateFeatures` section to manage things like cpu/memory hotplug .
+
+## Update/Rollback Compatibility
+
+The feautre doesn't intorduce any breaking changes so existing functionality of versions prior to this feature should
+not be impacted upon update/rollback. However if existing workloads would like to opt-in for this feature after an
+update, the VMI should be re-created since the addition of new fields to the VMI API.
+
+## Functional Testing Approach
+
+Functional tests will simply:
+
+* Create and start a VM with `memeory` set as one of th `liveUpdateFeatures`
+* Modify the number of memory guest for the VM
+* Ensure the guest sees the changed number of memory
+* Ensure the memory resources on the virt-launcher pod were adjusted accordingly
+* Ensure interoperability with non-hotplug related live-migration occasions (workload-update, evacuation,
```

Comment:
Can you elaborate on this point?

Reply:
Because the current implementation is dependent on migration, you need to consider some of the scenarios involved in migration

Reply:
Can you do an example?

Reply:
> Can you give an example?

Sorry, I want to say cpu hotplug is dependent on migration.